nixos/cinnamon: init

mkg20001 opened the pull request:
<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!
List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://hydra.nixos.org/job/nixpkgs/trunk/manual/latest/download/1/nixpkgs/manual.html#chap-reviewing-contributions
-->

###### Motivation for this change

Make it happen :tm: 

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

r-ryantm committed:
cinnamon.cinnamon-desktop: 4.4.1 -> 4.6.1

r-ryantm committed:
cinnamon.cinnamon-menus: 4.4.0 -> 4.6.0

r-ryantm committed:
cinnamon.cjs: 4.4.0 -> 4.6.0

mkg20001 committed:
arc-theme: enable cinnamon

worldofpeace commented on nixos/modules/services/x11/desktop-managers/cinnamon.nix:
```diff
@@ -0,0 +1,155 @@
+      source-code-pro # Default monospace font in 3.32
+      source-sans-pro
+    ];
+
```
you should be able to replace these both with
```
 services.xserver.displayManager.sessionPackages = [ 
  cinnamon.cinnamon-common
];
```

however, the xsession files aren't patched properly
```
[Desktop Entry]
Name=Cinnamon
Comment=This session logs you into Cinnamon
Exec=cinnamon-session-cinnamon
TryExec=/usr/bin/cinnamon
Icon=
Type=Application
```

The stream of execution requires patching like
1. the xsession files to`$out/bin`
2. $out/bin/cinnamon-session-cinnamon and $out/bin/cinnamon-session-cinnamon2d 
```
# these scripts need to be hardcoded to ${cinnamon-session}/bin/cinnamon-session
exec cinnamon-session --session cinnamon "$@"
```

worldofpeace commented on nixos/modules/services/x11/desktop-managers/cinnamon.nix:
```diff
@@ -0,0 +1,155 @@
+      # session requirements
+      cinnamon-screensaver
+      # nemo-autostart: provided by nemo
+      pkgs.gnome3.networkmanagerapplet
```
is it that we need the service running `programs.nm-applet.enable`. We're preferring to use this over the xdg-autostart in the package.

mkg20001 replied:
It was saying something about needing that or smth. Don't remember exactly but if nm-applet gets the job done than that works too

mkg20001 committed:
cinnamon.cinnamon-common: patch & expose sessions

mkg20001 committed:
cinnamon.cinnamon-control-center: 4.4.0 -> 4.6.0

mkg20001 committed:
cinnamon.cinnamon-screensaver: 4.4.0 -> 4.6.0

mkg20001 committed:
cinnamon.cinnamon-session: 4.4.1 -> 4.6.1

mkg20001 committed:
cinnamon.cinnamon-common: 4.4.1 -> 4.6.1

mkg20001 committed:
cinnamon.cinnamon-session: 4.4.0 -> 4.6.4

mkg20001 committed:
cinnamon.mint-themes: 1.8.0 -> 1.8.6

mkg20001 committed:
cinnamon.mint-y-icons: unstable -> 1.4.3

mkg20001 committed:
cinnamon.muffin: 4.4.2 -> 4.6.3

mkg20001 committed:
cinnamon.nemo: 4.4.1 -> 4.6.5

mkg20001 committed:
cinnamon.xapps: 1.6.10 -> 1.8.9

mkg20001 commented on nixos/modules/services/x11/desktop-managers/cinnamon.nix:
```diff
@@ -0,0 +1,148 @@
+      cinnamon-control-center
+      cinnamon-settings-daemon
+      gnome3.libgnomekbd
+      # TODO: these are optional, but part of "the cinnamon experience" - add via option like gnome? (also not packaged yet)
```
need an opinion on this

worldofpeace replied:
I think if they're soo few having an option isn't really worth it. Better to just use `removePackagesByName`

mkg20001 replied:
There's a lot more apps they made and other ones that are "shipped by default", I've' made it an option

mkg20001 committed:
cinnamon.cinnamon-common: fix nma missing

mkg20001 committed:
teams: add cinnamon team

mkg20001 committed:
cinnamon.*: now maintained by cinnamon team

worldofpeace commented on pkgs/desktops/cinnamon/mint-artwork/default.nix:
```diff
@@ -0,0 +1,25 @@
+  src = fetchurl {
+    url = "http://packages.linuxmint.com/pool/main/m/mint-artwork/mint-artwork_${version}.tar.xz";
+    sha256 = "126asxpg722qfg2wkwcr7bhsplchq3jn6bkdwf1scpc5za8dd62j";
+  };
```
you need to add `glib` to `nativeBuildInputs`. This is because there's a setup-hook that moves glib schemas to share/gsettings-schemas/${pname}-${version}/glib-2.0/schemas

mkg20001 committed:
mint-artwork: init at 1.4.3

mkg20001 committed:
cinnamon.cinnamon-gsettings-overrides: init

mkg20001 committed:
vanilla-dmz: 0.4.4 -> 0.4.5 (+license fix)

The package debian/copyright states Attribution-ShareAlike 3.0 Unported 
(CC BY-SA 3.0) which is cc-by-sa-30, was wrongly cc-by-nc-sa-30 which is 
unfree

mkg20001 committed:
cinnamon.mint-x-icons: init at 1.5.5

mkg20001 commented:
![image](https://user-images.githubusercontent.com/7735145/92328767-4d728200-f063-11ea-9683-dc18c545c891.png)
![image](https://user-images.githubusercontent.com/7735145/92328768-53686300-f063-11ea-9a7e-5fdea6d920d1.png)

worldofpeace commented on nixos/modules/services/x11/desktop-managers/cinnamon.nix:
```diff
@@ -0,0 +1,260 @@
+
+  config = mkMerge [
+    (mkIf (cfg.enable && config.services.xserver.displayManager.lightdm.enable && config.services.xserver.displayManager.lightdm.greeters.gtk.enable) {
+      services.xserver.displayManager.lightdm.greeters.gtk.extraConfig = mkDefault (builtins.readFile "${pkgs.cinnamon.mint-artwork}/etc/lightdm/lightdm-gtk-greeter.conf.d/99_linuxmint.conf");
```
I think this is a drop-in that you could just use environment.etc with. Plus, by doing mkDefault to extraConfig someone could discard the configuration.

mkg20001 replied:
I've made it specifically so people can override the config to their liking

worldofpeace replied:
Ah, I see.

worldofpeace commented on nixos/modules/services/x11/desktop-managers/cinnamon.nix:
```diff
@@ -0,0 +1,260 @@
+    (mkIf cfg.enable {
+      services.xserver.displayManager.sessionPackages = [ pkgs.cinnamon.cinnamon-common ];
+
+      services.xserver.displayManager.defaultSession = mkDefault "cinnamon";
```
Is this needed? The issue with having to do that in pantheon was a bug in the greeter.

worldofpeace commented on nixos/modules/services/x11/desktop-managers/cinnamon.nix:
```diff
@@ -0,0 +1,260 @@
+      hardware.pulseaudio.enable = mkDefault true;
+      security.polkit.enable = true;
+      services.accounts-daemon.enable = true;
+      services.bamf.enable = true;
+      services.tumbler.enable = mkDefault true;
```
it's unlikely u need this

mkg20001 replied:
```suggestion
      services.bamf.enable = true;
```

mkg20001 replied:
```suggestion
```

worldofpeace commented on nixos/modules/services/x11/desktop-managers/cinnamon.nix:
```diff
@@ -0,0 +1,260 @@
+      services.gnome3.glib-networking.enable = true;
+      services.gnome3.gnome-keyring.enable = true;
+      services.gvfs.enable = true;
+      services.gnome3.rygel.enable = mkDefault true;
```
is rygel used?

mkg20001 replied:
not really, what does it do

worldofpeace replied:
` A home media solution (UPnP AV MediaServer) that allows you to easily share audio, video and pictures to other devices`. I believe that's used in a sharing feature in gnome-control-center. You should probably check if/how sharing works for the cinnamon one.

worldofpeace commented on nixos/modules/services/x11/desktop-managers/cinnamon.nix:
```diff
@@ -0,0 +1,260 @@
+      services.gnome3.gnome-keyring.enable = true;
+      services.gvfs.enable = true;
+      services.gnome3.rygel.enable = mkDefault true;
+      services.gsignond.enable = mkDefault true;
+      services.gsignond.plugins = with pkgs.gsignondPlugins; [ lastfm mail oauth ];
```
these are only used in pantheon

mkg20001 replied:
```suggestion
```

worldofpeace commented on nixos/modules/services/x11/desktop-managers/cinnamon.nix:
```diff
@@ -0,0 +1,260 @@
+      services.upower.enable = mkDefault true; # config.powerManagement.enable;
+      services.xserver.libinput.enable = mkDefault true;
+      services.xserver.updateDbusEnvironment = true;
+      services.zeitgeist.enable = mkDefault true;
```
same with pantheon, not used here

mkg20001 replied:
```suggestion
```

worldofpeace commented on nixos/modules/services/x11/desktop-managers/cinnamon.nix:
```diff
@@ -0,0 +1,260 @@
+      services.zeitgeist.enable = mkDefault true;
+      services.geoclue2.enable = mkDefault true;
+      # Use gnome-settings-daemon fork
+      services.udev.packages = [
+        # pkgs.pantheon.elementary-settings-daemon
+      ];
+      systemd.packages = [
+        # pkgs.pantheon.elementary-settings-daemon
+      ];
```
```suggestion
```

worldofpeace commented on nixos/modules/services/x11/desktop-managers/cinnamon.nix:
```diff
@@ -0,0 +1,260 @@
+        gnome3.adwaita-icon-theme
+        gtk3.out
+        hicolor-icon-theme
+        lightlocker
```
not needed
```suggestion
```

worldofpeace commented on nixos/modules/services/x11/desktop-managers/cinnamon.nix:
```diff
@@ -0,0 +1,260 @@
+        lightlocker
+        nixos-artwork.wallpapers.simple-dark-gray
+        onboard
+        qgnomeplatform
```
it's up to you to use qgnomeplatform. but it's not necessary.

mkg20001 replied:
It makes qt stuff look like gtk stuff... sounds like a good feature

worldofpeace replied:
I actually believe, since the module was changed, we don't need to add `qgnomeplatform` to  systemPackages because it does this for you.

worldofpeace commented on nixos/modules/services/x11/desktop-managers/cinnamon.nix:
```diff
@@ -0,0 +1,260 @@
+
+        # session requirements
+        cinnamon-screensaver
+        # nemo-autostart: provided by nemo
+        pkgs.gnome3.networkmanagerapplet
```
```suggestion
```

I believe we figured out this wasn't needed (the  nixos service was though?)

worldofpeace commented on nixos/modules/services/x11/desktop-managers/cinnamon.nix:
```diff
@@ -0,0 +1,260 @@
+        gnome3.libgnomekbd
+
+        # theme
+        pkgs.gnome3.adwaita-icon-theme
+        pkgs.hicolor-icon-theme
+        pkgs.gnome3.gnome-themes-extra
+        pkgs.gtk3.out # for gtk-launch
```
some of this is duplicated from above

worldofpeace commented on nixos/modules/services/x11/desktop-managers/cinnamon.nix:
```diff
@@ -0,0 +1,260 @@
+        mint-y-icons
+        vanilla-dmz
+
+        # other
+        pkgs.glib # for gsettings
+        pkgs.shared-mime-info # for update-mime-database
+        pkgs.xdg-user-dirs
```
this was duplicated above

worldofpeace commented on nixos/modules/services/x11/desktop-managers/cinnamon.nix:
```diff
@@ -0,0 +1,260 @@
+      programs.evince.enable = mkDefault true;
+      programs.file-roller.enable = mkDefault true;
+
+      # Settings from elementary-default-settings
+      environment.sessionVariables.GTK_CSD = "1";
+      # environment.sessionVariables.GTK3_MODULES = [ "pantheon-filechooser-module" ];
```
```suggestion
```

worldofpeace commented on nixos/modules/services/x11/desktop-managers/cinnamon.nix:
```diff
@@ -0,0 +1,260 @@
+      # Override GSettings schemas
+      environment.sessionVariables.NIX_GSETTINGS_OVERRIDES_DIR = "${nixos-gsettings-desktop-schemas}/share/gsettings-schemas/nixos-gsettings-overrides/glib-2.0/schemas";
+
+      environment.sessionVariables.GNOME_SESSION_DEBUG = mkIf cfg.debug "1";
```
I'm not sure if cinnamon-session has a debug environment variable?

worldofpeace commented on nixos/modules/services/x11/desktop-managers/cinnamon.nix:
```diff
@@ -0,0 +1,260 @@
+        ubuntu_font_family # required for default theme
+      ];
+
+      /* fonts.fontconfig.defaultFonts = {
+        monospace = [ "Roboto Mono" ];
+        sansSerif = [ "Open Sans" ];
+      }; */
```
```suggestion
```

worldofpeace commented on pkgs/data/themes/arc/default.nix:
```diff
@@ -43,8 +44,8 @@ stdenv.mkDerivation rec {
   '';
 
   configureFlags = [
+    "--with-cinnamon=${cinnamon.cinnamon-common.version}"
```
This means gnome users of the theme will now have the cinnamon closure. Perhaps there should be split versions of a generic arc theme?

mkg20001 replied:
Unsure, how much problems does this actually cause?

romildo replied:
> This means gnome users of the theme will now have the cinnamon closure. Perhaps there should be split versions of a generic arc theme?

I do not think so. It is just telling what version of cinnamon should be targeted. In order to be added to the closure it should cinnamon should be added to `buildInputs`, right?

mkg20001 replied:
We're just pulling in version, so there's no way the whole thing gets added, I think

worldofpeace commented on pkgs/desktops/cinnamon/cinnamon-gsettings-schemas/default.nix:
```diff
@@ -0,0 +1,57 @@
+{ stdenv
```
preferably, you should name this `cinnamon-gsettings-overrides`. The name of the package in pantheon isn't correct and confusing.

worldofpeace commented on pkgs/desktops/cinnamon/cinnamon-screensaver/default.nix:
```diff
@@ -82,9 +84,7 @@ stdenv.mkDerivation rec {
   NIX_CFLAGS_COMPILE = "-I${glib.dev}/include/gio-unix-2.0"; # TODO: https://github.com/NixOS/nixpkgs/issues/36468
```
if we're using meson now, you should change this to being a mesonFlag (example is in the linked issue)

worldofpeace commented on pkgs/desktops/cinnamon/cinnamon-screensaver/default.nix:
```diff
@@ -82,9 +84,7 @@ stdenv.mkDerivation rec {
-    patchShebangs autogen.sh
-
-    sed ${stdenv.lib.escapeShellArg "s&DBUS_SESSION_SERVICE_DIR=.*&DBUS_SESSION_SERVICE_DIR=`$PKG_CONFIG --variable session_bus_services_dir dbus-1 | sed -e 's,/usr/share,\${datarootdir},g' | sed 's|^|$out|'`&g"} -i configure.ac
+    sed ${stdenv.lib.escapeShellArg "s|dbus_services_dir = .*|dbus_services_dir = 'share/dbus-1/services'|g"} -i meson.build
```
this `sed` is obsolete since the definition in the meson.build is supremely compatible  with nixos https://github.com/linuxmint/cinnamon-screensaver/blob/master/meson.build#L46

mkg20001 replied:
```
[0/1] Installing files.
Installing subdir /build/source/data/icons/hicolor to /nix/store/jbpibqbf8d9mzxp2fgv6pqgks0i26zvp-cinnamon-screensaver-4.6.0/share/icons/hicolor
Installing /build/source/data/icons/hicolor/scalable/actions/screensaver-unlock-symbolic.svg to /nix/store/jbpibqbf8d9mzxp2fgv6pqgks0i26zvp-cinnamon-screensaver-4.6.0/share/icons/hicolor/scalable/actions
Installing /build/source/data/icons/hicolor/scalable/actions/screensaver-switch-users-symbolic.svg to /nix/store/jbpibqbf8d9mzxp2fgv6pqgks0i26zvp-cinnamon-screensaver-4.6.0/share/icons/hicolor/scalable/actions
Installing /build/source/data/icons/hicolor/scalable/status/screensaver-blank.svg to /nix/store/jbpibqbf8d9mzxp2fgv6pqgks0i26zvp-cinnamon-screensaver-4.6.0/share/icons/hicolor/scalable/status
Installing /build/source/data/icons/hicolor/scalable/status/screensaver-notification-symbolic.svg to /nix/store/jbpibqbf8d9mzxp2fgv6pqgks0i26zvp-cinnamon-screensaver-4.6.0/share/icons/hicolor/scalable/status
Installing libcscreensaver/libcscreensaver.so.0.0.0 to /nix/store/jbpibqbf8d9mzxp2fgv6pqgks0i26zvp-cinnamon-screensaver-4.6.0/lib
Installing libcscreensaver/CScreensaver-1.0.gir to /nix/store/jbpibqbf8d9mzxp2fgv6pqgks0i26zvp-cinnamon-screensaver-4.6.0/share/gir-1.0
Installing libcscreensaver/CScreensaver-1.0.typelib to /nix/store/jbpibqbf8d9mzxp2fgv6pqgks0i26zvp-cinnamon-screensaver-4.6.0/lib/girepository-1.0
Installing src/pamhelper/cinnamon-screensaver-pam-helper to /nix/store/jbpibqbf8d9mzxp2fgv6pqgks0i26zvp-cinnamon-screensaver-4.6.0/libexec
Installing /build/source/libcscreensaver/cs-auth.h to /nix/store/jbpibqbf8d9mzxp2fgv6pqgks0i26zvp-cinnamon-screensaver-4.6.0/include/cinnamon-screensaver/libcscreensaver
Installing /build/source/libcscreensaver/setuid.h to /nix/store/jbpibqbf8d9mzxp2fgv6pqgks0i26zvp-cinnamon-screensaver-4.6.0/include/cinnamon-screensaver/libcscreensaver
Installing /build/source/build/meson-private/cscreensaver.pc to /nix/store/jbpibqbf8d9mzxp2fgv6pqgks0i26zvp-cinnamon-screensaver-4.6.0/lib/pkgconfig
Installing /build/source/data/cinnamon-screensaver to /nix/store/jbpibqbf8d9mzxp2fgv6pqgks0i26zvp-cinnamon-screensaver-4.6.0/etc/pam.d
Installing /build/source/build/data/cinnamon-screensaver.desktop to /nix/store/jbpibqbf8d9mzxp2fgv6pqgks0i26zvp-cinnamon-screensaver-4.6.0/share/applications
Traceback (most recent call last):
  File "/nix/store/cg1lyk9naby9pl4srl3igi5cca6qjzqc-meson-0.55.1/lib/python3.8/site-packages/mesonbuild/mesonmain.py", line 131, in run
    return options.run_func(options)
  File "/nix/store/cg1lyk9naby9pl4srl3igi5cca6qjzqc-meson-0.55.1/lib/python3.8/site-packages/mesonbuild/minstall.py", line 542, in run
    installer.do_install(datafilename)
  File "/nix/store/cg1lyk9naby9pl4srl3igi5cca6qjzqc-meson-0.55.1/lib/python3.8/site-packages/mesonbuild/minstall.py", line 365, in do_install
    self.install_data(d)
  File "/nix/store/cg1lyk9naby9pl4srl3igi5cca6qjzqc-meson-0.55.1/lib/python3.8/site-packages/mesonbuild/minstall.py", line 396, in install_data
    if self.do_copyfile(fullfilename, outfilename, makedirs=(d.dirmaker, outdir)):
  File "/nix/store/cg1lyk9naby9pl4srl3igi5cca6qjzqc-meson-0.55.1/lib/python3.8/site-packages/mesonbuild/minstall.py", line 263, in do_copyfile
    dirmaker.makedirs(outdir, exist_ok=True)
  File "/nix/store/cg1lyk9naby9pl4srl3igi5cca6qjzqc-meson-0.55.1/lib/python3.8/site-packages/mesonbuild/minstall.py", line 60, in makedirs
    os.makedirs(path, exist_ok=exist_ok)
  File "/nix/store/685kq8pyhrvajah1hdsfn4q7gm3j4yd4-python3-3.8.5/lib/python3.8/os.py", line 213, in makedirs
    makedirs(head, exist_ok=exist_ok)
  File "/nix/store/685kq8pyhrvajah1hdsfn4q7gm3j4yd4-python3-3.8.5/lib/python3.8/os.py", line 223, in makedirs
    mkdir(name, mode)
PermissionError: [Errno 13] Permission denied: '/etc/dbus-1'
FAILED: meson-install
```

worldofpeace replied:
this is because they're doing `define_variable` on `prefix` when it should be `datadir`

mkg20001 replied:
so we need a different sed then?

worldofpeace replied:
here's the patch https://github.com/linuxmint/cinnamon-screensaver/pull/349/files

worldofpeace commented on pkgs/desktops/cinnamon/cinnamon-screensaver/default.nix:
```diff
@@ -34,24 +35,25 @@
     wrapGAppsHook
     gettext
     intltool
     dbus # for configure.ac
```
for `meson.build`

worldofpeace commented on pkgs/desktops/cinnamon/xapps/default.nix:
```diff
@@ -77,16 +74,14 @@ stdenv.mkDerivation rec {
 
   postPatch = ''
-    chmod +x schemas/meson_install_schemas.py # patchShebangs requires executable file
+    sed ${stdenv.lib.escapeShellArg "/dbus_services_dir = .*/{n;d}"} -i meson.build
```
this should be obsolete now?

worldofpeace replied:
https://github.com/linuxmint/xapp/pull/110

worldofpeace commented on nixos/modules/services/x11/desktop-managers/cinnamon.nix:
```diff
@@ -0,0 +1,225 @@
+        cinnamon-screensaver = {};
+      };
+
+      environment.systemPackages = with pkgs.cinnamon // pkgs; pkgs.gnome3.removePackagesByName [
```
you shouldn't be using removePackagesByName on this set because you'll allow someone to break the session

worldofpeace commented on nixos/modules/services/x11/desktop-managers/cinnamon.nix:
```diff
@@ -0,0 +1,225 @@
+      };
+
+      environment.systemPackages = with pkgs.cinnamon // pkgs; pkgs.gnome3.removePackagesByName [
+        # things copied from pantheon
```
```suggestion
   
```

worldofpeace commented on nixos/modules/services/x11/desktop-managers/cinnamon.nix:
```diff
@@ -0,0 +1,225 @@
+      services.xserver.libinput.enable = mkDefault true;
+      services.xserver.updateDbusEnvironment = true;
+      services.geoclue2.enable = mkDefault true;
+      # Use gnome-settings-daemon fork
```
```suggestion
```

worldofpeace reviewed (changes requested):
We're missing configuration for `cinnamon-settings-daemon`.
I'm guessing it needs to be in ` services.udev.packages` `systemd.packages` and `systemPackages`.
Currently rebuilding the package to inspect its output.

worldofpeace commented:
`cinnamon-settings-daemon` could also do with a `dev` output.

romildo commented on pkgs/desktops/cinnamon/mint-x-icons/default.nix:
```diff
@@ -0,0 +1,57 @@
+
+stdenv.mkDerivation rec {
+  pname = "mint-x-icons";
+  version = "1.5.5";
```
`mint-x-icons` is already packaged in `pkgs/data/icons`. Maybe just move it `pkgs/desktop/cinnamon`.

mkg20001 committed:
mint-x-icons: remove other package and expose from cinnamon namespace

worldofpeace commented on nixos/modules/services/x11/desktop-managers/cinnamon.nix:
```diff
@@ -0,0 +1,226 @@
+        nemo
+        xapps
+      ];
+      services.cinnamon.apps.enable = mkDefault true;
```
```suggestion
```

already defined

worldofpeace commented on nixos/modules/services/x11/desktop-managers/cinnamon.nix:
```diff
@@ -0,0 +1,226 @@
+      services.gnome3.gnome-keyring.enable = true;
+      services.gvfs.enable = true;
+      services.udisks2.enable = true;
+      services.upower.enable = mkDefault true; # config.powerManagement.enable;
```
this shouldn't be commented like this?

mkg20001 replied:
which one should I pick, based on config.powerManagment or enable by default

worldofpeace replied:
everything in nixos seems work off `config.powerManagement.enable`

mkg20001 committed:
cinnamon.cinnamon-settings-daemon: add dev output

worldofpeace commented on nixos/modules/services/x11/desktop-managers/cinnamon.nix:
```diff
@@ -0,0 +1,223 @@
+    };
+
+  };
+
```
```suggestion
```

one line extra here

worldofpeace commented:
We also need to change the commit messages:
modules.cinnamon: add cinnamon module -> nixos/cinnamon: init
and everything that isn't already prefixed with a `cinnamon.*` attribute.

worldofpeace commented on nixos/modules/services/x11/desktop-managers/cinnamon.nix:
```diff
@@ -0,0 +1,214 @@
+
+      environment.systemPackages = with pkgs.cinnamon // pkgs; [
+        desktop-file-utils
+        glib
```
this is at the bottom

worldofpeace commented on nixos/modules/services/x11/desktop-managers/cinnamon.nix:
```diff
@@ -0,0 +1,214 @@
+        glib
+        nixos-artwork.wallpapers.simple-dark-gray
+        onboard
+        shared-mime-info
```
this is at the bottom

worldofpeace commented:
There's a path to the onboard theme in the mint-artwork override that isn't patched right
theme='/nix/store/bxv49xfvc0m5kxbrd03bsl4hj5xr0j5p-mint-artwork-1.4.3/share/mint-artwork/onboard/Nightshade.theme'

worldofpeace commented:
> There's a path to the onboard theme in the mint-artwork override that isn't patched right
> theme='/nix/store/bxv49xfvc0m5kxbrd03bsl4hj5xr0j5p-mint-artwork-1.4.3/share/mint-artwork/onboard/Nightshade.theme'
> that belongs to onboard

Turns out this does exist. It seems they changed the Nightshade theme from onboard in an almost imperceptible  way. I guess too little for us to care about.

mkg20001 committed:
nixos/cinnamon: init

Co-Authored-By: WORLDofPEACE <worldofpeace@protonmail.ch>
